packages/wallet/src/wallet.ts

```diff
@@ -327,4 +332,84 @@ export class Wallet {
         await this.db.saveSilentPaymentAddress(address);
         return address;
     }
+
+    public matchSilentBlockOutputs(
+        silentBlock: SilentBlock,
+        scanPrivateKey: Buffer,
+        scanPublicKey: Buffer,
+        spendPublicKey: Buffer,
+    ): Coin[] {
+        const matchedUTXOs: Coin[] = [];
+
+        for (const transaction of silentBlock.transactions) {
+            const outputs = transaction.outputs;
+
+            if (outputs.length === 0) continue;
+
+            const outputPubKeys = outputs.map((output) =>
+                Buffer.from('02' + output.pubKey, 'hex'),
+            );
+
+            const scanTweak = Buffer.from(transaction.scanTweak, 'hex');
+
+            const matches = scanOutputsWithTweak(
+                scanPrivateKey,
+                spendPublicKey,
+                scanTweak,
+                outputPubKeys,
+            );
+
+            if (matches.size === 0) continue;
+
+            for (const pubKeyHex of matches.keys()) {
+                const output = outputs.find(
+                    (output) => output.pubKey === pubKeyHex.slice(2),
+                );
+                if (output) {
+                    matchedUTXOs.push(
+                        new Coin({
+                            txid: transaction.txid,
+                            vout: output.vout,
+                            value: output.value,
+                            address: encodeSilentPaymentAddress(
+                                toXOnly(scanPublicKey),
+                                spendPublicKey,
+                                this.network.network,
+                            ),
```

Comment:
this is wrong, we need to store the address generated from the taproot key here
ie using the outputs from parsed silent block